File: frontend/src/types/helpBoard.ts
export type HelpCategory = "water" | "food" | "shelter" | "rides";
export type HelpPostStatus = "open" | "fulfilled";

export interface HelpPost {
  id: number;
  category: HelpCategory;
  title: string;
  description: string;
  location: string | null;
  contact_phone: string | null;
  status: HelpPostStatus;
  author_name: string;
  author_id: number;
  created_at: string;
  is_mine: boolean;
}

export interface HelpPostCreate {
  category: HelpCategory;
  title: string;
  description: string;
  location?: string;
  contact_phone?: string;
}

export const HELP_CATEGORIES: {
  id: HelpCategory;
  label: string;
  icon: string;
  color: string;
}[] = [
  { id: "water", label: "Water", icon: "💧", color: "cyan" },
  { id: "food", label: "Food", icon: "🍞", color: "amber" },
  { id: "shelter", label: "Shelter", icon: "🏠", color: "sky" },
  { id: "rides", label: "Rides", icon: "🚗", color: "violet" },
];

export function categoryMeta(category: HelpCategory) {
  return HELP_CATEGORIES.find((c) => c.id === category) ?? HELP_CATEGORIES[0];
}

File: backend/models/help_board.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class HelpCategory(str, Enum):
    water = "water"
    food = "food"
    shelter = "shelter"
    rides = "rides"


class HelpPostStatus(str, Enum):
    open = "open"
    fulfilled = "fulfilled"


class HelpPostCreate(BaseModel):
    category: HelpCategory
    title: str = Field(..., min_length=3, max_length=120)
    description: str = Field(..., min_length=10, max_length=1000)
    location: str | None = Field(default=None, max_length=200)
    contact_phone: str | None = Field(default=None, max_length=30)


class HelpPostOut(BaseModel):
    id: int
    category: HelpCategory
    title: str
    description: str
    location: str | None = None
    contact_phone: str | None = None
    status: HelpPostStatus
    author_name: str
    author_id: int
    created_at: datetime
    is_mine: bool = False

    model_config = {"from_attributes": True}
